ethcore/light/src/client/mod.rs

```diff
@@ -110,7 +112,11 @@ impl Client {
 		let best_num = self.chain.best_block().number;
 		self.tx_pool.lock()
 			.values()
-			.filter(|t| t.min_block.as_ref().map_or(true, |x| x <= &best_num))
+			.filter(|t| match t.condition {
+				Some(TransactionCondition::Number(ref x)) => x <= &best_num,
+				Some(TransactionCondition::Timestamp(ref x)) => *x <= time::get_time().sec as u64,
```

Comment:
this condition check needs updating as well. probably would be best done after changing `HeaderChain` to use `ethcore::encoded` (done locally).